k9mail-library/src/main/java/com/fsck/k9/mail/store/imap/selectedstate/command/UidFetchCommand.java

```diff
@@ -0,0 +1,147 @@
+package com.fsck.k9.mail.store.imap.selectedstate.command;
+
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.LinkedHashSet;
+import java.util.Locale;
+import java.util.Set;
+
+import com.fsck.k9.mail.BodyFactory;
+import com.fsck.k9.mail.FetchProfile;
+import com.fsck.k9.mail.K9MailLib;
+import com.fsck.k9.mail.Message;
+import com.fsck.k9.mail.MessagingException;
+import com.fsck.k9.mail.Part;
+import com.fsck.k9.mail.store.imap.Commands;
+import com.fsck.k9.mail.store.imap.FetchBodyCallback;
+import com.fsck.k9.mail.store.imap.FetchPartCallback;
+import com.fsck.k9.mail.store.imap.ImapConnection;
+import com.fsck.k9.mail.store.imap.ImapResponse;
+import com.fsck.k9.mail.store.imap.ImapResponseCallback;
+import com.fsck.k9.mail.store.imap.ImapUtility;
+
+
+public class UidFetchCommand extends SelectedStateCommand {
```

Comment:
Won't the current tests in `ImapFolderTest` cover this ? They either mock the `imapConnection.executeSimpleCommand` method to return mock responses when the correct command string is provided as an argument, or directly check if it was invoked with the correct command.